Backend/FcEvaluate.hs

```diff
@@ -96,11 +71,33 @@ smallStep (FcTmCaseFc scr alts) = smallStep scr >>= \case
   Just scr' -> pure $ Just $ FcTmCaseFc scr' alts
   Nothing   -> pure $ Nothing
 
+-- | Check whether a term is a user defined data constructor application.
+isUserDefinedDataConApp :: FcTerm 'Fc -> Bool
+isUserDefinedDataConApp = isJust . userDefinedDataConAppMaybe
+{-# INLINE isUserDefinedDataConApp #-}
+
+-- | Check whether a term is a user defined data constructor application
+-- | and return its parts.
```

Comment:
```suggestion
-- | Check whether a term is a user defined data constructor application
-- and return its parts.
```
General comment: starting the first line of a comment block with `-- |` is sufficient for getting haddock-compatible documentation. You don't have to add a bar on every line of the comment. :wink: